test/puppetlabs/ring_middleware/core_test.clj

```diff
@@ -463,3 +464,14 @@
         (is (= handled-response (wrapped-handler put-request)))
         (is (= handled-response (wrapped-handler post-request)))
         (is (= handled-response (wrapped-handler delete-request)))))))
+
+(deftest json-response-test
```

Comment:
This was the only easily extractable test from the ringutils tests that I found. Most of the testing for the functions pulled in with this PR were handled with integration test (eg. [file-sync's error handling tests](https://github.com/puppetlabs/pe-file-sync/blob/2016.2.x/test/integration/puppetlabs/enterprise/services/file_sync_storage/file_sync_storage_service_test.clj#L429-L513)). Which give me a high degree in the confidence of the PR thus far, though for maintainability it'd obviously be nice to have much more extensive testing in this PR.